dbutils.fs.ls('dbfs:/FileStore/tables/aula-databricks/')

/dbfs/FileStore/tables/aula-databricks/prouni/reajuste.csv

dbfs:/FileStore/tables/aula-databricks/prouni/cursos_prouni.csv

In [0]:
local_do_arquivo = '/FileStore/tables/aula-databricks/prouni/cursos_prouni.csv'

In [0]:
import pyspark.pandas as ps

df = ps.read_csv(local_do_arquivo)

In [0]:
df = ps.read_csv(local_do_arquivo, index_col='curso_id')

Medicina_no_Brasil

%md
No Brasil, os cursos de medicina são extremamente concorridos e exigentes, com uma carga horária intensa e duração média de seis anos. As faculdades de medicina no país oferecem formação em diversas áreas, como cirurgia, pediatria, ginecologia e obstetrícia, entre outras especialidades médicas.

Existem opções de bolsas para estudantes que desejam cursar medicina no país. Uma das principais opções é o Programa Universidade para Todos (PROUNI), que oferece bolsas de estudos parciais e integrais em universidades particulares para estudantes de baixa renda que tenham obtido boas notas no Enem.

Além disso, há também a possibilidade de ingressar em universidades públicas, que são gratuitas e oferecem um ensino de qualidade reconhecido mundialmente. No entanto, o processo seletivo para ingresso em universidades públicas é bastante concorrido e exige um alto desempenho dos candidatos.

Nas próximas seções, iremos disponibilizar alguns resultados com base em dados modificados do PROUNI e de universidades públicas, a fim de fornecer informações mais detalhadas sobre as opções de bolsas e ingresso nessas instituições de ensino.

In [0]:
df.head()

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 curso_busca 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15 
 14.0 
 NaN 
 NaN 
 Medicina 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 104191210567043 
 Bacharelado 
 Noturno 
 9836.40 
 1 
 NaN 
 NaN 
 NaN 
 Enfermagem 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Enfermagem 
 663.36 
 NaN 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2 
 5.0 
 6.0 
 10.0 
 Medicina 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 104191212798093 
 Bacharelado 
 Noturno 
 9689.34 
 3 
 2.0 
 NaN 
 NaN 
 Psicologia 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Psicologia 
 651.00 
 652.22 
 NaN 
 NaN 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4 
 1.0 
 5.0 
 2.0 
 Medicina 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26

In [0]:
df.shape

Out[314]: (41447, 19)

In [0]:
df['curso_busca'].equals(df['nome'])

Out[315]: curso_id
706710394154       True
104191210567043    True
1002328574024      True
104191212798093    True
65899611932754     True
65781564964        True
65788486984        True
65861111850254     True
658099184724       True
10505911785244     True
65921511844324     True
1055878921193      True
65881713994834     True
104123711716094    True
689693216461       True
6896933216461      True
105821211803184    True
699042726814       True
100369211074       True
6576768684         True
105887911921884    True
658119591744       True
105596711902804    True
25031009564        True
137380663614       True
405659244          True
256016294          True
699950009794       True
108119413996154    True
65941410292044     True
65886811637384     True
107783113635364    True
103635150009314    True
1004316634941      True
706703822244       True
1003235594144      True
2457651144         True
658670718594       True
105458350008124    True
702329994764       True
3349900844         Tr

In [0]:
df['curso_busca'].equals(df['nome']).sum()

Out[316]: 41447

In [0]:
#Excluir coluna
df = df.drop(['curso_busca'],axis=1)

In [0]:
#Renomear coluna
df = df.rename(columns={'nome':'nome_curso'})

In [0]:
df.head()

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15 
 14.0 
 NaN 
 NaN 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 104191210567043 
 Bacharelado 
 Noturno 
 9836.40 
 1 
 NaN 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Enfermagem 
 663.36 
 NaN 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2 
 5.0 
 6.0 
 10.0 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 104191212798093 
 Bacharelado 
 Noturno 
 9689.34 
 3 
 2.0 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Psicologia 
 651.00 
 652.22 
 NaN 
 NaN 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4 
 1.0 
 5.0 
 2.0 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26

In [0]:
#Quantos cursos de medicina estão disponíveis nesse dataset?

df[df['nome_curso']=='Medicina']

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15.0 
 14.0 
 NaN 
 NaN 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2.0 
 5.0 
 6.0 
 10.0 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4.0 
 1.0 
 5.0 
 2.0 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26 
 
 
 65781564964 
 Bacharelado 
 Integral 
 9650.00 
 8.0 
 15.0 
 NaN 
 NaN 
 Presidente Prudente 
 SP 
 MzUwODAzNjQxNDA2 
 Universidade do Oeste Paulista - UNOESTE 
 CAMPUS I SEDE ADMINISTRATIVA PRESIDENTE PRUDE... 
 657815 
 Medicina 
 738.92 
 728.86 
 NaN 
 NaN 
 
 
 65788486984 
 Bacharelado 
 Integral 
 9564.00 
 3.0 
 5.0 
 NaN 
 NaN 
 Marilia 
 SP 
 MzUwOTAzODI5MDA1 
 Universidade de Marília - UNIMAR 
 UNIDADE SEDE - Campus Universitario 
 657884 
 Medicina 
 744.16 
 735.00 
 NaN 
 NaN 
 
 
 65861111850254 
 Bacharelado 
 Integral 
 9350.00 
 4.0 
 3.0 
 NaN 
 NaN 
 Sao Joao Del Rei 
 MG 
 MzExMTA1ODYyNTAw 
 Centro Universitário Presidente Tancredo de Al... 
 UNIDADE SEDE (UNIDADE) - Fábricas 
 658611 
 Medicina 
 747.84 
 736.68 
 NaN 
 NaN 
 
 
 658099184724 
 Bacharelado 
 Integral 
 9208.00 
 7.0 
 5.0 
 NaN 
 NaN 
 Cuiaba 
 MT 
 NTEwNDAxNzAzNDAz 
 Universidade de Cuiabá - UNIC/UNIME 
 UNIC - BEIRA RIO (SEDE) - Jardim Europa 
 658099 
 Medicina 
 735.40 
 725.36 
 NaN 
 NaN 
 
 
 10505911785244 
 Bacharelado 
 Integral 
 9048.00 
 13.0 
 4.0 
 NaN 
 NaN 
 Lauro de Freitas 
 BA 
 MjkwNTAyMTE5MjA3 
 Faculdade de Ciências Agrárias e da Saúde - FAS 
 UNIDADE SEDE - Centro 
 105059 
 Medicina 
 741.04 
 730.36 
 NaN 
 NaN 
 
 
 65921511844324 
 Bacharelado 
 Integral 
 9023.07 
 4.0 
 4.0 
 5.0 
 5.0 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82 
 
 
 65881713994834 
 Bacharelado 
 Integral 
 8600.00 
 2.0 
 9.0 
 NaN 
 NaN 
 Rio Claro 
 SP 
 MzUwNjAyNjQzOTA3 
 Claretiano - Faculdade - Claretianorc - CLARET... 
 CLARETIANO - FACULDADE - Cidade Claret 
 658817 
 Medicina 
 734.44 
 720.86 
 NaN 
 NaN 
 
 
 104123711716094 
 Bacharelado 
 Integral 
 8508.54 
 2.0 
 1.0 
 7.0 
 2.0 
 Belem 
 PA 
 MTUwMzAwNzAxNDAy 
 Faculdade Metropolitana da Amazônia - FAMAZ 
 FACULDADE METROPOLITANA DA AMAZÔNIA - FAMAZ - ... 
 1041237 
 Medicina 
 741.36 
 739.90 
 760.80 
 715.10 
 
 
 105821211803184 
 Bacharelado 
 Integral 
 8402.00 
 4.0 
 1.0 
 NaN 
 NaN 
 Parnaiba 
 PI 
 MjIwMTAwMjA3NzAy 
 Faculdade de Ciências Humanas,Exatas e da Saúd... 
 CAMPUS PRINCIPAL / FAHESP- FACULDADE DE CIÊNCI... 
 1058212 
 Medicina 
 772.82 
 739.86 
 NaN 
 NaN 
 
 
 699042726814 
 Bacharelado 
 Integral 
 8350.00 
 4.0 
 3.0 
 NaN 
 NaN 
 Joao Pessoa 
 PB 
 MjUwNDAyMjA3NTA3 
 Faculdade de Medicina Nova Esperança - FAMENE 
 CAMPUS - JOÃO PESSOA - Gramame 
 699042 
 Medicina 
 759.76 
 736.94 
 NaN 
 NaN 
 
 
 100369211074 
 Bacharelado 
 Integral 
 8329.95 
 1.0 
 4.0 
 NaN 
 NaN 
 Caxias do Sul 
 RS 
 NDMwMjAxNjA1MTA4 
 Universidade de Caxias do Sul - UCS 
 CAMPUS SEDE - Petrópolis 
 1003692 
 Medicina 
 745.42 
 751.32 
 NaN 
 NaN 
 
 
 6576768684 
 Bacharelado 
 Integral 
 8290.00 
 0.0 
 7.0 
 NaN 
 NaN 
 C

In [0]:
df[df['nome_curso']=='Medicina']['nome_curso'].count()

Out[321]: 124

In [0]:
#Quais são os turnos do curso de Medicina?
df['turno'].unique()

Out[322]: 0             Matutino
1    Curso a Distância
2           Vespertino
3             Integral
4              Noturno
Name: turno, dtype: object

In [0]:
df[(df['turno']=='Integral') & (df['nome_curso']=='Medicina')]['nome_curso'].count()

Out[323]: 123

In [0]:
# | indica um operdor lógico OU
df[(df['turno']=='Integral') & (df['nome_curso']=='Medicina') | (df['turno']=='Matutino') & (df['nome_curso']=='Medicina') ]['nome_curso'].count()

Out[324]: 124

Reajuste das Mensalidades

In [0]:
local_do_arquivo = '/FileStore/tables/aula-databricks/prouni/reajuste.csv'
df_mensalidade = ps.read_csv(local_do_arquivo)

In [0]:
df_mensalidade.head()

reajuste 
 
 
 
 
 0 
 100 
 
 
 1 
 50 
 
 
 2 
 100 
 
 
 3 
 50 
 
 
 4 
 100

In [0]:
df_mensalidade.shape

Out[327]: (41447, 1)

In [0]:
# df['mensalidade'] = df['mensalidade'] + df_mensalidade['reajuste']

In [0]:
#Forma correta de executar a operação
ps.set_option('compute.ops_on_diff_frames',True)

In [0]:
df.reset_index(inplace=True)

In [0]:
df['mensalidade'] = df['mensalidade'] + df_mensalidade['reajuste']

Resumo do dataset

In [0]:
#Remoção dos dados NaN
df.dropna(inplace=True)
df.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 1950 
 65982912800701 
 Bacharelado 
 Matutino 
 1561.29 
 1 
 1 
 1 
 30 
 Caucaia 
 CE 
 MjMwMzAxNjAzNzA5 
 Faculdade Terra Nordeste - FATENE 
 UNIDADE SEDE - Soledade 
 659829 
 Engenharia Civil 
 688.70 
 693.02 
 450.00 
 450.00 
 
 
 8440 
 65928650009893 
 Bacharelado 
 Noturno 
 980.00 
 1 
 1 
 3 
 2 
 Sao Gotardo 
 MG 
 MzEwNTAyMDYyMTA0 
 Faculdade de Ciências Gerenciais de São Gotard... 
 UNIDADE SEDE - Boa Esperança 
 659286 
 Direito 
 638.94 
 612.92 
 608.26 
 579.50 
 
 
 9458 
 658024423323 
 Bacharelado 
 Noturno 
 927.00 
 2 
 1 
 3 
 2 
 Dourados 
 MS 
 NTAwNDAxMDAzNzAy 
 Centro Universitário da Grande Dourados - Unig... 
 DOURADOS - Jardim 
 658024 
 Psicologia 
 631.32 
 609.34 
 618.58 
 557.68 
 
 
 11190 
 1072630671593 
 Bacharelado 
 Noturno 
 829.00 
 6 
 5 
 10 
 9 
 Juiz de Fora 
 MG 
 MzExMjA2NTM2NzAy 
 Faculdade Doctum de Juiz de Fora - DOCTUM 
 CAMPUS APOGEU - Centro 
 1072630 
 Direito 
 669.40 
 632.94 
 615.56 
 584.58 
 
 
 17979 
 69558711774842 
 Tecnológico 
 Vespertino 
 543.00 
 1 
 1 
 1 
 1 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade da Amazônia Ocidental - FAAO 
 CAMPUS - RIO BRANCO - JARDIM PRIMAVERA - Jardi... 
 695587 
 Análise e Desenvolvimento de Sistemas 
 615.54 
 574.42 
 582.66 
 623.22

In [0]:
df.describe()

curso_id 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 campus_id 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 count 
 2.203000e+03 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 2.203000e+03 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 
 
 mean 
 4.376614e+13 
 881.605556 
 1.891058 
 2.082161 
 3.387199 
 7.190649 
 7.510660e+05 
 599.380390 
 574.097921 
 547.297204 
 521.452964 
 
 
 std 
 4.862265e+13 
 737.440291 
 1.969321 
 2.105756 
 4.020312 
 13.897084 
 4.202414e+05 
 53.194117 
 59.868058 
 58.971101 
 56.371352 
 
 
 min 
 1.649611e+08 
 194.000000 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 4.790000e+02 
 450.000000 
 450.000000 
 450.000000 
 450.000000 
 
 
 25% 
 6.589496e+11 
 500.000000 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 6.582050e+05 
 569.520000 
 542.220000 
 507.160000 
 450.000000 
 
 
 50% 
 1.002328e+13 
 773.830000 
 1.000000 
 1.000000 
 2.000000 
 3.000000 
 6.596950e+05 
 602.040000 
 583.320000 
 555.240000 
 526.540000 
 
 
 75% 
 1.040060e+14 
 1049.000000 
 2.000000 
 2.000000 
 4.000000 
 6.000000 
 1.054619e+06 
 634.600000 
 614.340000 
 588.160000 
 563.620000 
 
 
 max 
 2.008254e+14 
 9815.610000 
 34.000000 
 50.000000 
 74.000000 
 179.000000 
 2.008254e+06 
 781.200000 
 792.240000 
 776.360000 
 768.340000

In [0]:
df[df['nome_curso'] == 'Medicina'].describe()

curso_id 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 campus_id 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 count 
 1.500000e+01 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 1.500000e+01 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 
 
 mean 
 4.407875e+13 
 8007.354667 
 2.800000 
 2.266667 
 4.466667 
 3.866667 
 7.599417e+05 
 753.249333 
 738.438667 
 746.453333 
 724.676000 
 
 
 std 
 4.349636e+13 
 1096.173682 
 1.859339 
 1.533747 
 2.587516 
 2.669047 
 2.510458e+05 
 13.109076 
 16.966989 
 14.295313 
 18.302904 
 
 
 min 
 1.373807e+11 
 6113.000000 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 1.373800e+05 
 735.840000 
 720.200000 
 729.720000 
 690.180000 
 
 
 25% 
 6.584619e+11 
 7130.000000 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 6.583450e+05 
 741.360000 
 729.020000 
 735.760000 
 715.100000 
 
 
 50% 
 6.579365e+13 
 8295.000000 
 2.000000 
 1.000000 
 5.000000 
 3.000000 
 6.600790e+05 
 754.980000 
 737.880000 
 743.140000 
 721.080000 
 
 
 75% 
 7.070341e+13 
 8608.540000 
 4.000000 
 3.000000 
 6.000000 
 5.000000 
 1.041237e+06 
 762.740000 
 741.100000 
 750.440000 
 732.460000 
 
 
 max 
 1.058879e+14 
 9815.610000 
 8.000000 
 5.000000 
 9.000000 
 10.000000 
 1.079920e+06 
 781.200000 
 792.240000 
 776.360000 
 768.340000

In [0]:
df_descricao = df[df['nome_curso'] == 'Medicina'].describe()

In [0]:
# Nome das colunas para auxiliar ['mensalidade','curso_id','campus_id',"nota_integral_ampla", "nota_integral_cotas", "nota_parcial_ampla", "nota_parcial_cotas"]

In [0]:
df_descricao.drop(['mensalidade','curso_id','campus_id',"nota_integral_ampla", "nota_integral_cotas", "nota_parcial_ampla", "nota_parcial_cotas"],axis=1)

bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 
 
 
 
 count 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 
 
 mean 
 2.800000 
 2.266667 
 4.466667 
 3.866667 
 
 
 std 
 1.859339 
 1.533747 
 2.587516 
 2.669047 
 
 
 min 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 
 
 25% 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 
 
 50% 
 2.000000 
 1.000000 
 5.000000 
 3.000000 
 
 
 75% 
 4.000000 
 3.000000 
 6.000000 
 5.000000 
 
 
 max 
 8.000000 
 5.000000 
 9.000000 
 10.000000

In [0]:
df_descricao_bolsa = df_descricao.drop(['mensalidade','curso_id','campus_id',"nota_integral_ampla", "nota_integral_cotas", "nota_parcial_ampla", "nota_parcial_cotas"],axis=1)

In [0]:
df_descricao_bolsa

bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 
 
 
 
 count 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 
 
 mean 
 2.800000 
 2.266667 
 4.466667 
 3.866667 
 
 
 std 
 1.859339 
 1.533747 
 2.587516 
 2.669047 
 
 
 min 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 
 
 25% 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 
 
 50% 
 2.000000 
 1.000000 
 5.000000 
 3.000000 
 
 
 75% 
 4.000000 
 3.000000 
 6.000000 
 5.000000 
 
 
 max 
 8.000000 
 5.000000 
 9.000000 
 10.000000

In [0]:
#Podemos derrubar essa linha pois é apenas um contador
df_descricao_bolsa = df_descricao_bolsa.drop(['count'],axis=0)

In [0]:
#Estilzação do formato da tabela
#Quantidadade de fumero depois da vírgula
df_descricao_bolsa.style.format('{:,.2f}')

bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 
 
 
 
 mean 
 2.80 
 2.27 
 4.47 
 3.87 
 
 
 std 
 1.86 
 1.53 
 2.59 
 2.67 
 
 
 min 
 1.00 
 1.00 
 1.00 
 1.00 
 
 
 25% 
 1.00 
 1.00 
 1.00 
 2.00 
 
 
 50% 
 2.00 
 1.00 
 5.00 
 3.00 
 
 
 75% 
 4.00 
 3.00 
 6.00 
 5.00 
 
 
 max 
 8.00 
 5.00 
 9.00 
 10.00

In [0]:
#troca da cor do background
df_descricao_bolsa.style.format('{:,.2f}').background_gradient(axis=1)

bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 
 
 
 
 mean 
 2.80 
 2.27 
 4.47 
 3.87 
 
 
 std 
 1.86 
 1.53 
 2.59 
 2.67 
 
 
 min 
 1.00 
 1.00 
 1.00 
 1.00 
 
 
 25% 
 1.00 
 1.00 
 1.00 
 2.00 
 
 
 50% 
 2.00 
 1.00 
 5.00 
 3.00 
 
 
 75% 
 4.00 
 3.00 
 6.00 
 5.00 
 
 
 max 
 8.00 
 5.00 
 9.00 
 10.00

Qual é a média dos curso de medicina por estado?

In [0]:
df.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 1950 
 65982912800701 
 Bacharelado 
 Matutino 
 1561.29 
 1 
 1 
 1 
 30 
 Caucaia 
 CE 
 MjMwMzAxNjAzNzA5 
 Faculdade Terra Nordeste - FATENE 
 UNIDADE SEDE - Soledade 
 659829 
 Engenharia Civil 
 688.70 
 693.02 
 450.00 
 450.00 
 
 
 8440 
 65928650009893 
 Bacharelado 
 Noturno 
 980.00 
 1 
 1 
 3 
 2 
 Sao Gotardo 
 MG 
 MzEwNTAyMDYyMTA0 
 Faculdade de Ciências Gerenciais de São Gotard... 
 UNIDADE SEDE - Boa Esperança 
 659286 
 Direito 
 638.94 
 612.92 
 608.26 
 579.50 
 
 
 9458 
 658024423323 
 Bacharelado 
 Noturno 
 927.00 
 2 
 1 
 3 
 2 
 Dourados 
 MS 
 NTAwNDAxMDAzNzAy 
 Centro Universitário da Grande Dourados - Unig... 
 DOURADOS - Jardim 
 658024 
 Psicologia 
 631.32 
 609.34 
 618.58 
 557.68 
 
 
 11190 
 1072630671593 
 Bacharelado 
 Noturno 
 829.00 
 6 
 5 
 10 
 9 
 Juiz de Fora 
 MG 
 MzExMjA2NTM2NzAy 
 Faculdade Doctum de Juiz de Fora - DOCTUM 
 CAMPUS APOGEU - Centro 
 1072630 
 Direito 
 669.40 
 632.94 
 615.56 
 584.58 
 
 
 17979 
 69558711774842 
 Tecnológico 
 Vespertino 
 543.00 
 1 
 1 
 1 
 1 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade da Amazônia Ocidental - FAAO 
 CAMPUS - RIO BRANCO - JARDIM PRIMAVERA - Jardi... 
 695587 
 Análise e Desenvolvimento de Sistemas 
 615.54 
 574.42 
 582.66 
 623.22

In [0]:
df_medicina = df.query("nome_curso == 'Medicina'")
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82

In [0]:
df_medicina['mensalidade'].mean()

Out[345]: 8007.354666666666

In [0]:
#Mensalidade por cidade
# AND == &
df_medicina.query("turno == 'Integral' AND cidade_busca == 'Sao Paulo'")

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 2 
 1002328574024 
 Bacharelado 
 Integral 
 9815.61 
 2 
 5 
 6 
 10 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28

In [0]:
#Mensalidade por estado
df_medicina.query("turno == 'Integral' OR turno == 'Matutino'")

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82 
 
 
 120 
 658345863564 
 Bacharelado 
 Integral 
 6113.00 
 3 
 3 
 1 
 1 
 Vitoria 
 ES 
 MzIwMzAwOTA1MzA5 
 Faculdade Brasileira - MULTIVIX VITÓRIA 
 CAMPUS GOIABEIRAS - Goiabeiras 
 658345 
 Medicina 
 762.74 
 749.00 
 776.36 
 768.34 
 
 
 80 
 7052291076664 
 Bacharelado 
 Integral 
 7130.00 
 1 
 1 
 1 
 1 
 Montes Claros 
 MG 
 MzEwMjAwNzQzMzAy 
 Faculdades Integradas Pitágoras - FIP-MOC 
 UNIDADE SEDE - Ibituruna 
 705229 
 Medicina 
 765.76 
 741.10 
 774.96 
 750.64 
 
 
 2 
 1002328574024 
 Bacharelado 
 Integral 
 9815.61 
 2 
 5 
 6 
 10 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 4 
 65899611932754 
 Bacharelado 
 Integral 
 9774.34 
 4 
 1 
 5 
 2 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26 
 
 
 13 
 104123711716094 
 Bacharelado 
 Integral 
 8608.54 
 2 
 1 
 7 
 2 
 Belem 
 PA 
 MTUwMzAwNzAxNDAy 
 Faculdade Metropolitana da Amazônia - FAMAZ 
 FACULDADE METROPOLITANA DA AMAZÔNIA - FAMAZ - ... 
 1041237 
 Medicina 
 741.36 
 739.90 
 760.80 
 715.10 
 
 
 21 
 658119591744 
 Bacharelado 
 Integral 
 8381.37 
 4 
 1 
 8 
 2 
 Sao Luis 
 MA 
 MjEwMTAwMjExMzAw 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 UNICEUMA - CAMPUS RENASCENÇA - Renascença II 
 658119 
 Medicina 
 757.84 
 738.90 
 735.76 
 718.48 
 
 
 42 
 70703413994804 
 Bacharelado 
 Integral 
 7850.00 
 1 
 1 
 1 
 3 
 Campo Mourao 
 PR 
 NDEwMjAwNTA0MzAz 
 Centro Universitário Integrado de Campo Mourão 
 UNIDADE CAMPUS - Jardim Batel 
 707034 
 Medicina 
 745.46 
 746.10 
 743.14 
 690.18 
 
 
 20 
 105887911921884 
 Bacharelado 
 Integral 
 8381.38 
 3 
 1 
 6 
 2 
 Imperatriz 
 MA 
 MjEwMjAwOTA1MzAy 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 UNIVERSIDADE DO CEUMA - UNICEUMA - CAMPUS IMPE... 
 1058879 
 Medicina 
 756.46 
 737.88 
 747.14 
 717.60 
 
 
 60 
 1079920726674 
 Bacharelado 
 Integral 
 7512.70 
 8 
 5 
 5 
 3 
 Joao Pessoa 
 PB 
 MjUwNDAyMjA3NTA3 
 Faculdade de Ciências Médicas da Paraíba - FCM-PB 
 LADEIRA DE S FRANCISCO - Centro 
 1079920 
 Medicina 
 754.98 
 733.60 
 748.78 
 727.58 
 
 
 24 
 137380663614 
 Bacharelado 
 Integral 
 8295.00 
 1 
 1 
 4 
 3 
 Vespasiano 
 MG 
 MzEwNzAzMDcxMjA0 
 Faculdade da Saúde e Ecologia Humana - FASEH 
 UNIDADE

In [0]:
#Agrupando esses dados
agrupado = df_medicina.query("turno == 'Integral' OR turno == 'Matutino'").groupby('uf_busca')['mensalidade'].mean()

In [0]:
agrupado

Out[349]: uf_busca
RO    7009.970000
SP    8477.466667
ES    6113.000000
PB    7512.700000
MG    7712.500000
PA    8608.540000
RJ    9123.070000
MA    8381.375000
AC    9774.340000
PR    7174.275000
Name: mensalidade, dtype: float64

In [0]:
df_agrupado = ps.DataFrame(agrupado)

In [0]:
df_agrupado

mensalidade 
 
 
 uf_busca 
 
 
 
 
 
 RO 
 7009.970000 
 
 
 SP 
 8477.466667 
 
 
 ES 
 6113.000000 
 
 
 PB 
 7512.700000 
 
 
 MG 
 7712.500000 
 
 
 PA 
 8608.540000 
 
 
 RJ 
 9123.070000 
 
 
 MA 
 8381.375000 
 
 
 AC 
 9774.340000 
 
 
 PR 
 7174.275000

In [0]:
#ordenamento de informações´da cotegogia 'mensalidade'
df_agrupado.sort_values(by='mensalidade',inplace=True,ascending=False)

In [0]:
df_agrupado

mensalidade 
 
 
 uf_busca 
 
 
 
 
 
 AC 
 9774.340000 
 
 
 RJ 
 9123.070000 
 
 
 PA 
 8608.540000 
 
 
 SP 
 8477.466667 
 
 
 MA 
 8381.375000 
 
 
 MG 
 7712.500000 
 
 
 PB 
 7512.700000 
 
 
 PR 
 7174.275000 
 
 
 RO 
 7009.970000 
 
 
 ES 
 6113.000000

In [0]:
#É necessário que se ressete o index paea que se possa graficá-lo, uma ve que o íncice deve ser o eixo de estados do dataframe
#Gráfico de barras
df_agrupado = df_agrupado.reset_index()

In [0]:
#Agrupamento de forma decrescente
#Esssa é a biblioteca plotly
df_agrupado.plot.bar(x='uf_busca',y='mensalidade')

Qual é o número total de bolsas por estado?

ps.sql('''

''')

In [0]:
ps.sql('''
SELECT bolsa_integral_cotas, uf_busca
FROM {DF}
''', DF=df)

bolsa_integral_cotas 
 uf_busca 
 
 
 
 
 0 
 1 
 CE 
 
 
 1 
 1 
 MG 
 
 
 2 
 2 
 MS 
 
 
 3 
 6 
 MG 
 
 
 4 
 1 
 AC 
 
 
 5 
 1 
 PR 
 
 
 6 
 1 
 PR 
 
 
 7 
 1 
 PR 
 
 
 8 
 2 
 MT 
 
 
 9 
 1 
 SP 
 
 
 10 
 1 
 CE 
 
 
 11 
 1 
 PB 
 
 
 12 
 1 
 DF 
 
 
 13 
 1 
 CE 
 
 
 14 
 1 
 MG 
 
 
 15 
 1 
 SE 
 
 
 16 
 1 
 PR 
 
 
 17 
 1 
 SP 
 
 
 18 
 4 
 GO 
 
 
 19 
 1 
 PR 
 
 
 20 
 6 
 SP 
 
 
 21 
 1 
 MT 
 
 
 22 
 1 
 AC 
 
 
 23 
 1 
 ES 
 
 
 24 
 1 
 SP 
 
 
 25 
 1 
 DF 
 
 
 26 
 1 
 PR 
 
 
 27 
 1 
 SE 
 
 
 28 
 1 
 SP 
 
 
 29 
 1 
 PR 
 
 
 30 
 1 
 RO 
 
 
 31 
 1 
 RJ 
 
 
 32 
 1 
 SP 
 
 
 33 
 2 
 MT 
 
 
 34 
 2 
 MG 
 
 
 35 
 2 
 SP 
 
 
 36 
 2 
 RR 
 
 
 37 
 2 
 ES 
 
 
 38 
 2 
 MG 
 
 
 39 
 1 
 PB 
 
 
 40 
 1 
 SP 
 
 
 41 
 4 
 ES 
 
 
 42 
 4 
 SP 
 
 
 43 
 1 
 GO 
 
 
 44 
 2 
 GO 
 
 
 45 
 1 
 SP 
 
 
 46 
 3 
 BA 
 
 
 47 
 5 
 BA 
 
 
 48 
 3 
 PA 
 
 
 49 
 1 
 ES 
 
 
 50 
 1 
 SP 
 
 
 51 
 1 
 AL 
 
 
 52 
 3 
 SP 
 
 
 53 
 2 
 GO 
 
 
 54 
 2 
 MS 
 
 
 55 
 1 
 SP 
 
 
 56 
 2 
 SP 
 
 
 57 
 1 
 PB 
 
 
 58 
 1 
 PR 
 
 
 59 
 1 
 MG 
 
 
 60 
 1 
 GO 
 
 
 61 
 1 
 SP 
 
 
 62 
 1 
 GO 
 
 
 63 
 1 
 PR 
 
 
 64 
 2 
 SP 
 
 
 65 
 3 
 CE 
 
 
 66 
 2 
 SP 
 
 
 67 
 4 
 RO 
 
 
 68 
 1 
 SP 
 
 
 69 
 15 
 MA 
 
 
 70 
 5 
 MA 
 
 
 71 
 1 
 MG 
 
 
 72 
 1 
 BA 
 
 
 73 
 2 
 MG 
 
 
 74 
 1 
 SP 
 
 
 75 
 1 
 SP 
 
 
 76 
 2 
 SP 
 
 
 77 
 1 
 DF 
 
 
 78 
 1 
 BA 
 
 
 79 
 1 
 RJ 
 
 
 80 
 2 
 SP 
 
 
 81 
 1 
 CE 
 
 
 82 
 1 
 PR 
 
 
 83 
 1 
 SP 
 
 
 84 
 2 
 ES 
 
 
 85 
 1 
 PB 
 
 
 86 
 2 
 SP 
 
 
 87 
 3 
 PI 
 
 
 88 
 8 
 MS 
 
 
 89 
 1 
 SP 
 
 
 90 
 1 
 BA 
 
 
 91 
 2 
 BA 
 
 
 92 
 1 
 TO 
 
 
 93 
 2 
 MG 
 
 
 94 
 2 
 ES 
 
 
 95 
 2 
 PE 
 
 
 96 
 5 
 MG 
 
 
 97 
 1 
 MT 
 
 
 98 
 1 
 SP 
 
 
 99 
 4 
 RO 
 
 
 100 
 1 
 PR 
 
 
 101 
 1 
 MT 
 
 
 102 
 1 
 SP 
 
 
 103 
 1 
 PR 
 
 
 104 
 1 
 SP 
 
 
 105 
 2 
 SP 
 
 
 106 
 1 
 DF 
 
 
 107 
 1 
 SP 
 
 
 108 
 5 
 SP 
 
 
 109 
 1 
 DF 
 
 
 110 
 5 
 ES 
 
 
 111 
 1 
 AL 
 
 
 112 
 2 
 PI 
 
 
 113 
 1 
 PR 
 
 
 114 
 2 
 ES 
 
 
 115 
 1 
 BA 
 
 
 116 
 2 
 PR 
 
 
 117 
 1 
 CE 
 
 
 118 
 1 
 SP 
 
 
 119 
 1 
 SP 
 
 
 120 
 2 
 MS 
 
 
 121 
 2 
 SP 
 
 
 122 
 2 
 SP 
 
 
 123 
 1 
 SP 
 
 
 124 
 1 
 PR 
 
 
 125 
 2 
 SP 
 
 
 126 
 1 
 MS 
 
 
 127 
 1 
 PR 
 
 
 128 
 1 
 PR 
 
 
 129 
 2 
 ES 
 
 
 130 
 1 
 PR 
 
 
 131 
 1 
 MG 
 
 
 132 
 2 
 ES 
 
 
 133 
 4 
 AM 
 
 
 134 
 2 
 ES 
 
 
 135 
 1 
 PR 
 
 
 136 
 1 
 PR 
 
 
 137 
 2 
 MS 
 
 
 138 
 1 
 SP 
 
 
 139 
 1 
 GO 
 
 
 140 
 1 
 MS 
 
 
 141 
 1 
 GO 
 
 
 142 
 1 
 MG 
 
 
 143 
 1 
 PR 
 
 
 144 
 1 
 GO 
 
 
 145 
 1 
 SE 
 
 
 146 
 2 
 RN 
 
 
 147 
 1 
 SP 
 
 
 148 
 2 
 CE 
 
 
 149 
 1 
 SP 
 
 
 150 
 1 
 SP 
 
 
 151 
 1 
 BA 
 
 
 152 
 2 
 ES 
 
 
 153 
 1 
 CE 
 
 
 154 
 1 
 PI 
 
 
 155 
 4 
 ES 
 
 
 156 
 1 
 MG 
 
 
 157 
 4 
 RO 
 
 
 158 
 1 
 SP 
 
 
 159 
 2 
 ES 
 
 
 160 
 3 
 AC 
 
 
 161 
 1 
 GO 
 
 
 162 
 2 
 BA 
 
 
 163 
 2 
 GO 
 
 
 164 
 1 
 SP 
 
 
 165 
 2 
 MS 
 
 
 166 
 1 
 SP 
 
 
 167 
 2 
 PR 
 
 
 168 
 1 
 SP 
 
 
 169 
 2 
 ES 
 
 
 170 
 1 
 SP 
 
 
 171 
 1 
 SP 
 
 
 172 
 2 
 MA 
 
 
 173 
 1 
 CE 
 
 
 174 
 1 
 ES 
 
 
 175 
 1 
 RS 
 
 
 176 
 1 
 MG 
 
 
 177 
 1 
 MG 
 
 
 178 
 2 
 MG 
 
 
 179 
 2 
 SP 
 
 
 180 
 1 
 ES 
 
 
 181 
 2 
 MT 
 
 
 182 
 1 
 MT 
 
 
 183 
 1 
 SP 
 
 
 184 
 1 
 ES 
 
 
 185 
 4 
 PR 
 
 
 186 
 2 
 MA 
 
 
 187 
 1 
 SP 
 
 
 188 
 2 
 MS 
 
 
 189 
 1 
 SP 
 
 
 190 
 1 
 PR 
 
 
 191 
 1 
 SP 
 
 
 192 
 2 
 MS 
 
 
 193 
 1 
 SP 
 
 
 194 
 1 
 MT 
 
 
 195 
 1 
 PR 
 
 
 196 
 2 
 ES 
 
 
 197 
 1 
 SP 
 
 
 198 
 4 
 MS 
 
 
 199 
 1 
 RO 
 
 
 200 
 1 
 SP 
 
 
 201 
 1 
 SP 
 
 
 202 
 2 
 MS 
 
 
 203 
 1 
 PE 
 
 
 204 
 5 
 SP 
 
 
 205 
 1 
 AC 
 
 
 206 
 1 
 SP 
 
 
 207 
 1 
 SP 
 
 
 208 
 1 
 SP 
 
 
 209 
 1 
 ES 
 
 
 210 
 1 
 SP 
 
 
 211 
 1 
 MG 
 
 
 212 
 1 
 SP 
 
 
 213 
 3 
 MG 
 
 
 2

In [0]:
ps.sql('''
SELECT bolsa_integral_cotas, uf_busca
FROM {DF}
WHERE nome_curso = 'Medicina'
LIMIT 5
''', DF=df)

bolsa_integral_cotas 
 uf_busca 
 
 
 
 
 0 
 2 
 SP 
 
 
 1 
 2 
 SP 
 
 
 2 
 4 
 RO 
 
 
 3 
 1 
 PR 
 
 
 4 
 4 
 RJ

cotas_ordenadas = ps.sql('''

''')

In [0]:
cotas_ordenadas = ps.sql('''
SELECT uf_busca, SUM(bolsa_integral_cotas) AS Total_de_cotas
FROM {DF}
WHERE nome_curso='Medicina'
GROUP BY uf_busca
ORDER BY Total_de_cotas DESC
''', DF=df)

In [0]:
cotas_ordenadas.head()

uf_busca 
 Total_de_cotas 
 
 
 
 
 0 
 PB 
 8 
 
 
 1 
 MA 
 7 
 
 
 2 
 SP 
 6 
 
 
 3 
 RO 
 4 
 
 
 4 
 RJ 
 4

In [0]:
cotas_ordenadas.plot.bar(x='uf_busca',y='Total_de_cotas')

Temos mais cotas em capitais de estado ou em outros municípios?

In [0]:
df_medicina['cidade_busca'].unique()

Out[361]: 0          Sao Luis
1       Joao Pessoa
2             Belem
3         Itaperuna
4           Vitoria
5            Franca
6         Sao Paulo
7        Guarapuava
8      Campo Mourao
9     Montes Claros
10       Rio Branco
11       Imperatriz
12       Vespasiano
13      Porto Velho
Name: cidade_busca, dtype: object

In [0]:
#Uso do método MAP -> Criação de um dicionário
nomes_cidades = {
"Sao Luis": "Capital",
"Joao Pessoa":"Capital",
"Belem":"Capital",
"Itaperuna":"Município",
"Vitoria":"Município",
"Franca":"Município",
"Sao Paulo":"Capital",
"Guarapuava":"Município",
"Campo Mourao":"Município",
"Montes Claros":"Município",
"Rio Branco":"Município",
"Imperatriz":"Município",
"Vespasiano":"Município",
"Porto Velho": "Município"
}

In [0]:
df_medicina['cidade_busca'].map(nomes_cidades)

Out[363]: 22       Capital
71     Município
87     Município
107    Município
10     Município
120    Município
80     Município
2        Capital
4      Município
13       Capital
21       Capital
42     Município
20     Município
60       Capital
24     Município
Name: cidade_busca, dtype: object

In [0]:
capitais = df_medicina['cidade_busca'].map(nomes_cidades)

In [0]:
#Inserção de informaçãoes no df
df_medicina.insert(loc=8,column='tipo_cidade',value=capitais)

In [0]:
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 tipo_cidade 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Capital 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Município 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Município 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Município 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Município 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82

cotas_ordenadas = ps.sql('''

''')

In [0]:
cotas_ordenadas = ps.sql('''
SELECT tipo_cidade, SUM(bolsa_integral_cotas) AS Total_de_cotas
FROM {DF}
GROUP BY tipo_cidade
ORDER BY Total_de_cotas DESC
''', DF=df_medicina)

In [0]:
cotas_ordenadas = ps.sql('''
SELECT tipo_cidade, SUM(bolsa_integral_cotas) AS Total_de_cotas
FROM {DF}
GROUP BY tipo_cidade
ORDER BY Total_de_cotas DESC
''', DF=df_medicina)
cotas_ordenadas.plot.bar(x='tipo_cidade',y='Total_de_cotas')

Como a mensalidade do curso se relaciona com a quantidade de bolsas?

In [0]:
aux = df_medicina[['mensalidade','bolsa_integral_cotas','bolsa_integral_ampla','bolsa_parcial_cotas','bolsa_parcial_ampla','nota_integral_ampla', 'nota_integral_cotas', 'nota_parcial_ampla','nota_parcial_cotas']]

In [0]:
#Contrução de uma matrix de correlção
#Esse comando indica apenas correlações fortes positivas
aux.corr()

/databricks/spark/python/pyspark/sql/pandas/conversion.py:122: UserWarning:

toPandas attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, failed by the reason below:
  Unable to convert the field pearson(__correlation_output__). If this column is not necessary, you may consider dropping it or converting to primitive type before the conversion.
Direct cause: Unsupported type in conversion to Arrow: MatrixUDT()
Attempting non-optimization as 'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to true.



mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 mensalidade 
 1.000000 
 0.080124 
 0.085915 
 0.462838 
 0.281507 
 -0.438041 
 -0.042471 
 -0.490010 
 -0.556348 
 
 
 bolsa_integral_cotas 
 0.080124 
 1.000000 
 0.520982 
 0.451340 
 -0.120903 
 -0.155176 
 -0.360329 
 -0.247502 
 -0.064000 
 
 
 bolsa_integral_ampla 
 0.085915 
 0.520982 
 1.000000 
 0.182384 
 0.620009 
 -0.489539 
 -0.293681 
 -0.218251 
 0.084619 
 
 
 bolsa_parcial_cotas 
 0.462838 
 0.451340 
 0.182384 
 1.000000 
 0.195821 
 -0.468888 
 -0.253827 
 -0.546129 
 -0.469743 
 
 
 bolsa_parcial_ampla 
 0.281507 
 -0.120903 
 0.620009 
 0.195821 
 1.000000 
 -0.549894 
 -0.270162 
 -0.534464 
 -0.266389 
 
 
 nota_integral_ampla 
 -0.438041 
 -0.155176 
 -0.489539 
 -0.468888 
 -0.549894 
 1.000000 
 0.420823 
 0.409216 
 0.563932 
 
 
 nota_integral_cotas 
 -0.042471 
 -0.360329 
 -0.293681 
 -0.253827 
 -0.270162 
 0.420823 
 1.000000 
 0.402048 
 0.237963 
 
 
 nota_parcial_ampla 
 -0.490010 
 -0.247502 
 -0.218251 
 -0.546129 
 -0.534464 
 0.409216 
 0.402048 
 1.000000 
 0.702974 
 
 
 nota_parcial_cotas 
 -0.556348 
 -0.064000 
 0.084619 
 -0.469743 
 -0.266389 
 0.563932 
 0.237963 
 0.702974 
 1.000000

In [0]:
aux.corr().style.background_gradient(cmap="Reds",vmin=-1,vmax=1)

/databricks/spark/python/pyspark/sql/pandas/conversion.py:122: UserWarning:

toPandas attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, failed by the reason below:
  Unable to convert the field pearson(__correlation_output__). If this column is not necessary, you may consider dropping it or converting to primitive type before the conversion.
Direct cause: Unsupported type in conversion to Arrow: MatrixUDT()
Attempting non-optimization as 'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to true.



mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 mensalidade 
 1.000000 
 0.080124 
 0.085915 
 0.462838 
 0.281507 
 -0.438041 
 -0.042471 
 -0.490010 
 -0.556348 
 
 
 bolsa_integral_cotas 
 0.080124 
 1.000000 
 0.520982 
 0.451340 
 -0.120903 
 -0.155176 
 -0.360329 
 -0.247502 
 -0.064000 
 
 
 bolsa_integral_ampla 
 0.085915 
 0.520982 
 1.000000 
 0.182384 
 0.620009 
 -0.489539 
 -0.293681 
 -0.218251 
 0.084619 
 
 
 bolsa_parcial_cotas 
 0.462838 
 0.451340 
 0.182384 
 1.000000 
 0.195821 
 -0.468888 
 -0.253827 
 -0.546129 
 -0.469743 
 
 
 bolsa_parcial_ampla 
 0.281507 
 -0.120903 
 0.620009 
 0.195821 
 1.000000 
 -0.549894 
 -0.270162 
 -0.534464 
 -0.266389 
 
 
 nota_integral_ampla 
 -0.438041 
 -0.155176 
 -0.489539 
 -0.468888 
 -0.549894 
 1.000000 
 0.420823 
 0.409216 
 0.563932 
 
 
 nota_integral_cotas 
 -0.042471 
 -0.360329 
 -0.293681 
 -0.253827 
 -0.270162 
 0.420823 
 1.000000 
 0.402048 
 0.237963 
 
 
 nota_parcial_ampla 
 -0.490010 
 -0.247502 
 -0.218251 
 -0.546129 
 -0.534464 
 0.409216 
 0.402048 
 1.000000 
 0.702974 
 
 
 nota_parcial_cotas 
 -0.556348 
 -0.064000 
 0.084619 
 -0.469743 
 -0.266389 
 0.563932 
 0.237963 
 0.702974 
 1.000000

In [0]:
#Plot do gráfico de disperção
df_medicina.plot.scatter(x='mensalidade',y='bolsa_parcial_cotas')

In [0]:
#Ajuste de uma linha de tendências
#O excesso de disprssão indica a baixa correl
df_medicina.plot.scatter(x='mensalidade',y='bolsa_parcial_cotas', trendline='ols', trendline_color_override = 'red')

In [0]:
df_medicina.plot.scatter(x='nota_parcial_ampla',y='nota_parcial_cotas', trendline='ols', trendline_color_override = 'red')

Como o turno do curso afeta a mensalidade?

In [0]:
#comparação com boxplot
df['mensalidade'].plot.box()

In [0]:
df[df['turno']=='Integral']['mensalidade'].plot.box()

In [0]:
df[df['turno']=='Matutino']['mensalidade'].plot.box()

In [0]:
df[df['nome_curso']=='Medicina']['mensalidade'].plot.box()

In [0]:
df[df['nome_curso']=='Enfermagem']['mensalidade'].plot.box()

In [0]:
df[df['nome_curso']=='Enfermagem']['mensalidade'].plot.box(fillcolor='red')

Quais faculdades são universidades?

In [0]:
#Para pegar valores únicos
df_medicina['universidade_nome'].unique()

Out[381]: 0             Faculdades Integradas Pitágoras - FIP-MOC
1          Faculdade da Saúde e Ecologia Humana - FASEH
2                          Faculdade das Américas - FAM
3           Faculdade Metropolitana da Amazônia - FAMAZ
4     Faculdade de Ciências Médicas da Paraíba - FCM-PB
5                      Faculdade Redentor - FACREDENTOR
6                   Faculdade Barão do Rio Branco - FAB
7             Universidade Cidade de São Paulo - UNICID
8        Centro Universitário Integrado de Campo Mourão
9                     Faculdade Campo Real - CAMPO REAL
10               Centro Universitário São Lucas - UniSL
11              Faculdade Brasileira - MULTIVIX VITÓRIA
12          Universidade do Ceuma - Uniceuma - UNICEUMA
13                     Universidade de Franca - UNIFRAN
Name: universidade_nome, dtype: object

In [0]:
# .str -> método string
# .contains -> buscador
selecao = df_medicina[df_medicina['universidade_nome'].str.contains('Universidade')][['universidade_nome']]

In [0]:
selecao

universidade_nome 
 
 
 
 
 2 
 Universidade Cidade de São Paulo - UNICID 
 
 
 20 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 
 
 21 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 
 
 71 
 Universidade de Franca - UNIFRAN

Como pegar só a sigla das faculdades em geral?

In [0]:
#Inclusão de um centro universitário e criação de uma sigla, para manter o padrão
df_medicina['universidade_nome'].loc[42] = 'Centro Universitário Integrado de Campo Mourão - CUICM'
df_medicina['universidade_nome'].loc[42]

Out[384]: 'Centro Universitário Integrado de Campo Mourão - CUICM'

In [0]:
#Dataframe que separa o nome da sigla da universidade
#.str.split -> comando de separação
separado = df_medicina['universidade_nome'].str.split(' - ',n=1,expand=True)
separado

0 
 1 
 
 
 
 
 22 
 Faculdade das Américas 
 FAM 
 
 
 71 
 Universidade de Franca 
 UNIFRAN 
 
 
 87 
 Centro Universitário São Lucas 
 UniSL 
 
 
 107 
 Faculdade Campo Real 
 CAMPO REAL 
 
 
 10 
 Faculdade Redentor 
 FACREDENTOR 
 
 
 120 
 Faculdade Brasileira 
 MULTIVIX VITÓRIA 
 
 
 80 
 Faculdades Integradas Pitágoras 
 FIP-MOC 
 
 
 2 
 Universidade Cidade de São Paulo 
 UNICID 
 
 
 4 
 Faculdade Barão do Rio Branco 
 FAB 
 
 
 13 
 Faculdade Metropolitana da Amazônia 
 FAMAZ 
 
 
 21 
 Universidade do Ceuma 
 Uniceuma - UNICEUMA 
 
 
 42 
 Centro Universitário Integrado de Campo Mourão 
 CUICM 
 
 
 20 
 Universidade do Ceuma 
 Uniceuma - UNICEUMA 
 
 
 60 
 Faculdade de Ciências Médicas da Paraíba 
 FCM-PB 
 
 
 24 
 Faculdade da Saúde e Ecologia Humana 
 FASEH

In [0]:
df_medicina.insert(loc=12,column='sigla',value=separado[1])

In [0]:
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 tipo_cidade 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 sigla 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Capital 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 FAM 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Município 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 UNIFRAN 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Município 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 UniSL 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Município 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 CAMPO REAL 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Município 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 FACREDENTOR 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82

Qual a nota média de cada faculdade?

In [0]:
notas_ordenadas = ps.sql('''
SELECT sigla, MEAN(nota_integral_ampla) AS nota_media
FROM {DF}
GROUP BY sigla
ORDER BY nota_media DESC
''', DF=df_medicina)
notas_ordenadas.plot.bar(x='sigla',y='nota_media')

https://pt.wikipedia.org/wiki/Programa_Universidade_para_Todos

In [0]:
import pandas as pd

In [0]:
#Dados retirados do Wikipedia
#Confecção de um dicionário
data = {
    'Ano': [2021,2020],
    'Bolsas_Integrais': [146.3, 182.9],
    'Bolsas_Parciais': [150.0, 237.4],
    'Bolsas_Total': [296.3, 420.3]
}

In [0]:
#utilizar a biblioteca Plotly
df_pandas = pd.DataFrame(data)

In [0]:
df_bolsas = ps.from_pandas(df_pandas)
df_bolsas

Ano 
 Bolsas_Integrais 
 Bolsas_Parciais 
 Bolsas_Total 
 
 
 
 
 0 
 2021 
 146.3 
 150.0 
 296.3 
 
 
 1 
 2020 
 182.9 
 237.4 
 420.3

In [0]:
df_bolsas.plot.bar(x='Ano',y=['Bolsas_Integrais','Bolsas_Parciais'],barmode='group')

Como puxar informações de um site?

In [0]:
url = 'https://pt.wikipedia.org/wiki/Ensino_superior_no_Brasil'

In [0]:
#Bibliteca responsavel pela leitura de html
!pip install lxml

You should consider upgrading via the '/local_disk0/.ephemeral_nfs/envs/pythonEnv-adf825df-d057-419b-9cc5-42f44fc309bf/bin/python -m pip install --upgrade pip' command.


In [0]:
lista = ps.read_html(url)

In [0]:
len(lista)

Out[397]: 5

In [0]:
lista[0]

Classificação Nacional 
 Classificação Mundial 
 Universidade 
 Logo 
 sigla 
 tipo 
 web 
 Localização 
 
 
 
 
 0 
 1 
 72 
 Universidade de São Paulo 
 NaN 
 USP 
 Pública Estadual 
 www5.usp.br/ 
 São Paulo 
 
 
 1 
 2 
 236 
 Universidade Estadual de Campinas 
 NaN 
 UNICAMP 
 Pública Estadual 
 www.unicamp.br/ 
 São Paulo 
 
 
 2 
 3 
 278 
 Universidade Federal do Rio de Janeiro 
 NaN 
 UFRJ 
 Pública federal 
 ufrj.br/ 
 Rio de Janeiro 
 
 
 3 
 4 
 298 
 Universidade Federal de Minas Gerais 
 NaN 
 UFMG 
 Pública federal 
 ucb.edu.bo 
 Minas Gerais 
 
 
 4 
 5 
 345 
 Universidade Estadual Paulista 
 NaN 
 UNESP 
 Pública Estadual 
 www.unesp.br 
 São Paulo 
 
 
 5 
 6 
 427 
 Universidade Federal de Santa Catarina 
 NaN 
 UFSC 
 Pública federal 
 ufsc.br/ 
 Santa Catarina 
 
 
 6 
 7 
 592 
 Universidade de Brasília 
 NaN 
 UNB 
 Pública Estadual 
 www.unb.br/ 
 Distrito Federal 
 
 
 7 
 8 
 593 
 Universidade Federal do Paraná 
 NaN 
 UFPR 
 Pública federal 
 www.ufpr.br/ 
 Paraná 
 
 
 8 
 9 
 596 
 Universidade Federal Fluminense 
 NaN 
 UFF 
 Pública federal 
 www.uff.br/ 
 Rio de Janeiro 
 
 
 9 
 10 
 667 
 Universidade Federal de São Carlos 
 NaN 
 UFSCAR 
 Pública federal 
 www.ufscar.br/ 
 São Paulo

In [0]:
df_publicas = lista[0]
df_publicas

Classificação Nacional 
 Classificação Mundial 
 Universidade 
 Logo 
 sigla 
 tipo 
 web 
 Localização 
 
 
 
 
 0 
 1 
 72 
 Universidade de São Paulo 
 NaN 
 USP 
 Pública Estadual 
 www5.usp.br/ 
 São Paulo 
 
 
 1 
 2 
 236 
 Universidade Estadual de Campinas 
 NaN 
 UNICAMP 
 Pública Estadual 
 www.unicamp.br/ 
 São Paulo 
 
 
 2 
 3 
 278 
 Universidade Federal do Rio de Janeiro 
 NaN 
 UFRJ 
 Pública federal 
 ufrj.br/ 
 Rio de Janeiro 
 
 
 3 
 4 
 298 
 Universidade Federal de Minas Gerais 
 NaN 
 UFMG 
 Pública federal 
 ucb.edu.bo 
 Minas Gerais 
 
 
 4 
 5 
 345 
 Universidade Estadual Paulista 
 NaN 
 UNESP 
 Pública Estadual 
 www.unesp.br 
 São Paulo 
 
 
 5 
 6 
 427 
 Universidade Federal de Santa Catarina 
 NaN 
 UFSC 
 Pública federal 
 ufsc.br/ 
 Santa Catarina 
 
 
 6 
 7 
 592 
 Universidade de Brasília 
 NaN 
 UNB 
 Pública Estadual 
 www.unb.br/ 
 Distrito Federal 
 
 
 7 
 8 
 593 
 Universidade Federal do Paraná 
 NaN 
 UFPR 
 Pública federal 
 www.ufpr.br/ 
 Paraná 
 
 
 8 
 9 
 596 
 Universidade Federal Fluminense 
 NaN 
 UFF 
 Pública federal 
 www.uff.br/ 
 Rio de Janeiro 
 
 
 9 
 10 
 667 
 Universidade Federal de São Carlos 
 NaN 
 UFSCAR 
 Pública federal 
 www.ufscar.br/ 
 São Paulo

In [0]:
type(df_publicas)

Out[400]: pyspark.pandas.frame.DataFrame

In [0]:
#Para ordenar as informações de forma decrescente/descendente
df_publicas.sort_values(by=['Classificação Nacional'],inplace=True, ascending=False)

In [0]:
fig = df_publicas.plot.bar(x='sigla',y='Classificação Nacional')
fig.show()

In [0]:
#Aplicando gradiente de coloração para dar mais enfase a melhor posição em dentrimento da classificação mundial
fig = df_publicas.plot.bar(x='sigla',y='Classificação Nacional',color=range(10))
fig.show()

In [0]:
fig = df_publicas.plot.bar(x='sigla',y='Classificação Nacional',color=range(10))
fig.layout.coloraxis.showscale = False
fig.show()

Funcionalidades da PySpark

In [0]:
#Transformar Pandas e PySpark
dfs_bolsas = df_bolsas.to_spark()

In [0]:
dfs_bolsas.show()

+----+----------------+---------------+------------+
| Ano|Bolsas_Integrais|Bolsas_Parciais|Bolsas_Total|
+----+----------------+---------------+------------+
|2021|           146.3|          150.0|       296.3|
|2020|           182.9|          237.4|       420.3|
+----+----------------+---------------+------------+



Observação: no comando dfs_bolsas = df_bolsas.to_spark(), teremos a mensagem de que não há uma coluna de índice passada além da coluna convencional de 0 a 1.

Essa informação é levantada ao fazer a passagem do dataframe da Pandas ou da Pandas do Spark para o dataframe no formato do Spark.

In [0]:
type(dfs_bolsas)

Out[407]: pyspark.sql.dataframe.DataFrame

In [0]:
local_do_arquivo = '/FileStore/cursos_prouni.csv'

In [0]:
local_do_arquivo = '/FileStore/tables/cursos_prouni.csv'

In [0]:
#Fomato para salvar BigData -> .parquet e não .csv
#é um formato de partição de dados em pastas menores
df_publicas.to_parquet('/FileStore/tables/publicas.parquet',partition_cols='Classificação Nacional')

In [0]:
ps.read_parquet('/FileStore/tables/publicas.parquet')

Classificação Mundial 
 Universidade 
 Logo 
 sigla 
 tipo 
 web 
 Localização 
 Classificação Nacional 
 
 
 
 
 0 
 236 
 Universidade Estadual de Campinas 
 NaN 
 UNICAMP 
 Pública Estadual 
 www.unicamp.br/ 
 São Paulo 
 2 
 
 
 1 
 667 
 Universidade Federal de São Carlos 
 NaN 
 UFSCAR 
 Pública federal 
 www.ufscar.br/ 
 São Paulo 
 10 
 
 
 2 
 427 
 Universidade Federal de Santa Catarina 
 NaN 
 UFSC 
 Pública federal 
 ufsc.br/ 
 Santa Catarina 
 6 
 
 
 3 
 278 
 Universidade Federal do Rio de Janeiro 
 NaN 
 UFRJ 
 Pública federal 
 ufrj.br/ 
 Rio de Janeiro 
 3 
 
 
 4 
 298 
 Universidade Federal de Minas Gerais 
 NaN 
 UFMG 
 Pública federal 
 ucb.edu.bo 
 Minas Gerais 
 4 
 
 
 5 
 596 
 Universidade Federal Fluminense 
 NaN 
 UFF 
 Pública federal 
 www.uff.br/ 
 Rio de Janeiro 
 9 
 
 
 6 
 345 
 Universidade Estadual Paulista 
 NaN 
 UNESP 
 Pública Estadual 
 www.unesp.br 
 São Paulo 
 5 
 
 
 7 
 592 
 Universidade de Brasília 
 NaN 
 UNB 
 Pública Estadual 
 www.unb.br/ 
 Distrito Federal 
 7 
 
 
 8 
 593 
 Universidade Federal do Paraná 
 NaN 
 UFPR 
 Pública federal 
 www.ufpr.br/ 
 Paraná 
 8 
 
 
 9 
 72 
 Universidade de São Paulo 
 NaN 
 USP 
 Pública Estadual 
 www5.usp.br/ 
 São Paulo 
 1

Criando um dashboard

#Exercício de coloração do background

import pyspark.pandas as ps

# Crie um DataFrame de exemplo
data = {
    'App': ['App_A', 'App_B', 'App_C', 'App_D', 'App_E'],
    'Avaliação': [3.5, 4.2, 3.8, 4.5, 2.9],
    'Downloads': [15000, 23000, 18000, 27000, 10000]
}

df = ps.DataFrame(data)

df[['App', 'Avaliação', 'Downloads']].style.background_gradient(axis=0)

df[['Avaliação', 'Downloads']].style.background_gradient(axis=0)